### ISSUE TITLE
[Feature Request] Completion is not suggesting yet undefined (workspace/temporary) variables

### ISSUE THREAD

**Issue by astares** [guillep/pharo-core#13760]
Open a Playground and write some doIt code with an already defined temporary variable. The completion will honor that and suggest the variable like in this example:

<img width="206" alt="image" src="https://github.com/pharo-project/pharo/assets/5980033/61c10c00-b5ff-4493-8b03-45a5b173b731">

But often you just write the code without having yet declared the variable as a temporary variable or a workspace variable. In such a case the auto-completion does not suggest the variable yet:

<img width="160" alt="image" src="https://github.com/pharo-project/pharo/assets/5980033/52aedc13-47f5-413d-ae3e-7f5f32c0d79d">

Maybe also in the second example the completion could already suggest to complete the (upcoming) variable

**Comment by MarcusDenker** [comment]
Not yet defined variables it can not suggest as they do not exist yet. (they used to be colored red, maybe we should bring  that back?)

**Comment by guillep** [comment]
Why a "not yet defined variable"? In the context of an evaluator widget, it has a local binding, so we can consider it as always implicitly defined.

**Comment by MarcusDenker** [comment]
But the local binding only gets created the first time you execute, before that it just does not exist, so we can not really look it up.
Maybe we have to find a way to declare it earlier, but when? We e.g. hightlight at every keystroke, so we would create a lot of invalid vars if we would create them when highlighting.

**Comment by guillep** [comment]
Yes, I agree, I think that the evaluator widget should collaborate more closely with the compiler and parser then. It should not be much of a black box. The evaluator should call the compiler, ask for the bindings...
